src/viam/sdk/robot/client.hpp

```diff
@@ -81,54 +130,47 @@ class RobotClient {
     ///
     /// Because the return type here is a `Resource`, the user will need to manually
     /// cast to the desired type.
-    std::shared_ptr<Resource> resource_by_name(const ResourceName& name);
+    std::shared_ptr<Resource> resource_by_name(const Name& name);
 
     template <typename T>
     /// @brief Lookup and return a `shared_ptr` to a resource of the requested type.
     /// @param name The ordinary name of the resource.
     /// @throws `std::runtime_error` if the requested resource doesn't exist or is the wrong type.
     /// @return a `shared_ptr` to the requested resource.
     std::shared_ptr<T> resource_by_name(std::string name) {
-        ResourceName r;
-        API api = API::get<T>();
-        *r.mutable_namespace_() = api.type_namespace();
-        *r.mutable_type() = api.resource_type();
-        *r.mutable_subtype() = api.resource_subtype();
-        *r.mutable_name() = std::move(name);
-
-        auto resource = this->resource_by_name(std::move(r));
+        auto resource = this->resource_by_name({API::get<T>(), "", name});
         return std::dynamic_pointer_cast<T>(resource);
     }
 
     /// @brief Get the configuration of the frame system of the given robot.
     /// @return The configuration of the calling robot's frame system.
-    std::vector<FrameSystemConfig> get_frame_system_config(
-        std::vector<Transform> additional_transforms = std::vector<Transform>());
+    std::vector<frame_system_config> get_frame_system_config(
+        std::vector<WorldState::transform> additional_transforms =
+            std::vector<WorldState::transform>());
```

Comment:
You can make this simpler: `std::vector<WorldState::transform> additional_transforms = {}`